crates/optimism/trie/src/live.rs

```diff
@@ -17,31 +17,31 @@ use tracing::debug;
 
 /// Live trie collector for external proofs storage.
 #[derive(Debug)]
-pub struct LiveTrieCollector<Node, PreimageStore>
+pub struct LiveTrieCollector<Evm, Provider, PreimageStore>
 where
-    Node: FullNodeComponents,
-    Node::Provider: StateReader + DatabaseProviderFactory + StateProviderFactory,
+    Evm: ConfigureEvm,
+    Provider: StateReader + DatabaseProviderFactory + StateProviderFactory,
 {
-    evm_config: Node::Evm,
-    provider: Node::Provider,
+    evm_config: Evm,
+    provider: Provider,
     storage: PreimageStore,
 }
 
-impl<Node, Store, Primitives> LiveTrieCollector<Node, Store>
+impl<Evm, Provider, Store> LiveTrieCollector<Evm, Provider, Store>
 where
-    Node: FullNodeComponents<Types: NodeTypes<Primitives = Primitives>>,
-    Primitives: NodePrimitives,
+    Evm: ConfigureEvm,
+    Provider: StateReader + DatabaseProviderFactory + StateProviderFactory,
     Store: OpProofsStorage + Clone + 'static,
 {
     /// Create a new `LiveTrieCollector` instance
-    pub const fn new(evm_config: Node::Evm, provider: Node::Provider, storage: Store) -> Self {
+    pub const fn new(evm_config: Evm, provider: Provider, storage: Store) -> Self {
         Self { evm_config, provider, storage }
     }
 
     /// Execute a block and store the updates in the storage.
     pub async fn execute_and_store_block_updates(
         &self,
-        block: &RecoveredBlock<Primitives::Block>,
+        block: &RecoveredBlock<<Evm::Primitives as NodePrimitives>::Block>,
```

Comment:
```suggestion
        block: &RecoveredBlock<BlockTy<Evm::Primitives>>,
```
use `reth_primitives_traits::BlockTy`